ballerina/modules/parser/parser.bal

```diff
@@ -342,6 +342,17 @@ public class Parser {
         return alias;
     }
 
+    isolated function getTypeIdentifierTokenValue(Token token) returns string|Error {
+        string varType = check getIdentifierTokenvalue(token);
+        Token nextToken = check self.peekNextNonSeparatorToken();
+        if nextToken.kind == T_EXCLAMATION {
```

Comment:
What happens when we find square brackets? Don't we need to handle that?
Names with square brackets are considered in the `getOfTypeNameFromTypeName` function, but I think in this state if the document has a square bracket, it will result in a syntax error.

A better way to handle this is to throw an error stating that the lists are not yet supported.

We also have to be careful about not closing square brackets, which should give a syntax error. 
We should also consider the scenarios like `$info: [Information!]!`, where the list and the element type both are non-nullable types and other permutations of these.